octodns_gcore/__init__.py

```diff
@@ -368,6 +380,47 @@ def _data_for_CAA(self, _type, record):
             ],
         }
 
+    def _data_dynamic_healthcheck(self, record):
+
+        check_params = record['meta']['failover']
+
+        defaults = [
+            resource['content'][0] for resource in record["resource_records"]
+        ]
+
+        pools = {
+            'pool-0': {
+                'values': [
+                    {
+                        'value': resource['content'][0],
+                        'status': 'obey',
+                        # 'status': 'up' if resource['enabled'] else 'down',
+                    }
+                    for resource in record["resource_records"]
+                ]
+            }
+        }
+        octodns = {
+            'healthcheck': {
+                'host': check_params['host'],
+                # path properties is associated with url gcore param
+                'path': check_params['url'],
+                # 'url': check_params['url'],
+                'port': check_params['port'],
+                'protocol': check_params['protocol'],
+                'frequency': check_params['frequency'],
```

Comment:
This doesn't appear to be an `octodns.healthcheck` field. Any "extensions" that are provider specific should live under a provider specific key, e.g. for Route53's interval:

https://github.com/octodns/octodns-route53/blob/d9b046b2de2a8db6809efd914a624035af577a00/octodns_route53/provider.py#L1366-L1371

Any other provider-specific fields here should do something similar.

Reply:
Ok. I'm going to take inspiration from Route53 implementation.